src/components/manifold-subscription-create/manifold-subscription-create.tsx

```diff
@@ -196,15 +226,21 @@ export class ManifoldSubscriptionCreate {
       this.calculatedCost = cost;
     } catch (e) {
       if (e.name !== 'AbortError') {
-        // TODO store error in a better way so it can be displayed in place of cost in UI
-        this.errors = [{ message: 'Error getting plan cost.' }];
+        this.addErrors(dataError('cost', 'cost of selected plan'));
       }
```

Comment:
this should catch 4xx/5xx errors but it doesn't. `gateway.post` should probably be refactored to match the `graphqlFetch` API. so we can do this:
```js
const { data, errors } = await this.connection.gateway.post(...);
```

Reply:
I think `gateway.post` is only swallowing Network errors, and it shouldn't do that.

https://github.com/manifoldco/manifold-init/blob/master/src/v0/gateway.ts#L44

Here, if we've exhausted retries, we should be throwing a `ManifoldError` like `graphqlFetch` does https://github.com/manifoldco/manifold-init/blob/master/src/v0/graphqlFetch.ts#L126


Looks like it's not retrying on 401s either :disappointed:

Reply:
https://github.com/manifoldco/engineering/issues/11028